### PR TITLE
Version 1

### PR DESCRIPTION
**Pull request by MahdiAghaali**
## In this branch I and my partner @Efiamotu-1  did the following activities and changes:

- follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
    - with Devise.
- Food list:
    - display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
    - lead to a form that allows users to add new food.
- Recipes list:
    - display a list of recipes created by the logged-in user as in the wireframe.
    - lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
    - display a list of all public recipes ordered by newest as in the wireframe.
    - lead to recipe details.
- Recipe details:
    - If the recipe is public or the user is the owner of the recipe, display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, lead to the form that allows the user to add new food.
- General shopping list view (**only for a group that has less than 3 members**):
    - show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
    - count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @MahdiAghaali, @Efiamotu-1 🙋‍♀️,  

While you made a great effort in this project, I can't review your code.

## Invalid Code Review Request
 
You have submitted a project that has a README that is not professional. Here are key points to consider when documenting your README.

- The template logo should be removed. Optionally, the Microverse logo can be replaced with your own project logo if you have one. ✔️
- The only optional section is the FAQ section. If there's no information in the FAQ section then it should be deleted. ✔️ 
- All main headers from the [README template](https://github.com/microverseinc/readme-template) should be present. Even if there's no information for a section or sub-section, rather leave texts like "NA", "None" or "coming soon" than delete it. ✖️
***You are missing the `Acknowledgements` section kindly restore it. You can either thank the Microverse team for the great curriculum, and coding partners or the morning session team for support and motivation.***
- Future Features section should include 1-3 potential features to be added. ✖️  
-  Each section description should be removed and replaced with their custom content. The section descriptions are meant to guide you in documenting your README. They are not meant to be part of the final documentation. ✖️

image

- The License suggestion should be removed. ✔️ 
- There should not be any blank sections that do not have content. ✖️ 

Kindly document the `Key Features` section and list [1-3] key features in your application.
image
Kindly document the ` Future Features` section and list [1-3] future features you wish to implement in the future.
image


- There should not be any broken links in the README file. ✖️ 
***Delete the live demo section if the application is not deployed. If project requirements say to deploy the project, the section should be restored and the link updated.***

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/recipe_foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  def create
+    foods = Food.find_by(name: recipe_foods_params[:food])
+    recipe = Recipe.find(params[:recipe_id])
+    ingredient = RecipeFood.new(food: foods, recipe:, quantity: recipe_foods_params[:quantity])
+
+    if ingredient.save
+      flash[:success] = 'Ingredient added'
+    else
+      flash[:alert] = 'Something went wrong'
+    end
```
Comment (bobb-Rob) [inline app/controllers/recipe_foods_controller.rb:11]: - Your **General shopping list view**
    * Should show the list of food that is missing for all recipes of the logged-in user.
    * Should count the total food items and total price of the missing food.
   
/Foods - _list of foods_
image

_/shopping_list_

image

/recipe/2 - _recipe details_
image

On the shopping_list view, the quantity should be the difference from what you have already in the recipe detail view. example -> you had `87kg` in the food list and in the recipe you had `123kg`, so the shopping list should show `123kg - 87kg`. Meaning that is the quantity that should be purchased.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi Team, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — APPROVED** [review]
Hi @TEAM, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**